misk/src/main/kotlin/misk/grpc/GrpcChannel.kt

```diff
@@ -0,0 +1,43 @@
+package misk.grpc
+
+import java.io.Closeable
+import java.util.concurrent.LinkedBlockingDeque
+
+interface GrpcSendChannel<T> : Closeable {
+  fun send(message: T)
+}
+
+interface GrpcReceiveChannel<T> {
+  fun receiveOrNull(): T?
+}
+
+fun <T> GrpcReceiveChannel<T>.consumeEach(block: (T) -> Unit) {
+  while (true) {
+    val message = receiveOrNull() ?: return
+    block(message)
+  }
+}
+
+class BlockingGrpcChannel<T> : GrpcSendChannel<T>, GrpcReceiveChannel<T> {
+  private val queue = LinkedBlockingDeque<T>(2)
```

Comment:
2 feels hacky but this class won't survive the week so I'm okay with it.

Reply:
It works with 1 yo
so I changed it to 1